### ISSUE TITLE
Add the possibility to use multiple GPUs

### ISSUE THREAD

**Issue by alkamid** [beniz/deepdetect#179]
Caffe can use multiple GPUs for training. Could we have this feature in deepdetect as well?
I suppose the lines to modify would be  https://github.com/beniz/deepdetect/blob/3861bcff31a633c9efacb38d265186553de560bc/src/caffelib.cc#L1461? Something similar to https://github.com/BVLC/caffe/blob/master/tools/caffe.cpp#L247? If `gpuid` is to remain an `int`, maybe `-1` could mean "all GPUs"?

**Comment by beniz** [comment]
PR #192 adds support for multiple GPUs with Caffe.
The following range of API `gpuid` parameter types and values are supported:

- single value, e.g. `gpuid:0` or `gpuid:1`
- special `gpuid:-1` value to specify that all GPUs should be used
- any combination in an array, e.g. `gpuid:[1,3]` to use two GPUs among many for instance

**Comment by beniz** [comment]
Merged.

**Comment by supaggregator** [comment]
Hello, I have tried to set the gpuid parameter but it looks like not accepting array. 
Then I tries a specific gpuid number to create the prediction service(from pretrained model), the server will not follow it (all created on gpuid 0) and when I call the prediction service (with gpuid other than 0), I got a 1007 deepdetect error (something like src/caffe/layers/cudnn_conv_layer.cu:28 ). Does it has a reference with CUDA version (I use CUDA 7.5)? Thanks a lot.
